sendinblue.php

```diff
@@ -2993,6 +2996,222 @@ public function mailSendBySmtp()
 
 		return $this->_html_smtp_tracking;
 	}
+	/**
+	* Our newsletter blocks display and hide in Front Office.
+	*/
+	public function foBlockPosition()

```

Comment:
This file starts to be really big, and it would be better to separate the controllers and the views. Typically, this function generates HTML, which is wrong.
It should instead preparing all the data (i.e, the $module_data, the hook list etc), and THEN display your template with all this data.

You can find the documentation here:
http://doc.prestashop.com/display/PS15/Creating+a+PrestaShop+module#CreatingaPrestaShopmodule-UsingSmarty

I won't ask you to change the whole module for this time, but please, make at least this function respecting the MVC structure. Your module will be easier to read. Thanks